src/Popper.js

```diff
@@ -114,71 +125,60 @@ export class InnerPopper extends React.Component<PopperProps, PopperState> {
     !this.state.data ? undefined : this.state.data.placement;
 
   getArrowStyle = () =>
-    !this.state.arrowNode || !this.state.data
+    !this.arrowNode || !this.state.data
       ? initialArrowStyle
       : this.state.data.arrowStyles;
 
   getOutOfBoundariesState = () =>
     this.state.data ? this.state.data.hide : undefined;
 
-  initPopperInstance = () => {
-    const { referenceElement } = this.props;
-    const { popperNode, popperInstance } = this.state;
-    if (referenceElement && popperNode && !popperInstance) {
-      const popperInstance = new PopperJS(
-        referenceElement,
-        popperNode,
-        this.getOptions()
-      );
-      this.setState({ popperInstance });
-      return true;
-    }
-    return false;
+  destroy = () => {
+    if (!this.popperInstance) return false;
+    this.popperInstance.destroy();
+    this.popperInstance = null;
+    return true;
   };
 
-  destroyPopperInstance = (callback: () => boolean) => {
-    if (this.state.popperInstance) {
-      this.state.popperInstance.destroy();
-    }
-    this.setState({ popperInstance: undefined }, callback);
+  destroyPopperInstance = (callback: () => void) => {
```

Comment:
I'm a bit confused by the `destroy` vs `destroyPopperInstance` methods, the latter also forces a React re-render, but I don't understand how do you decide when to use one rather than the other. (Also, we could give them more explicit names maybe)

Reply:
The reason for both is this. When destorying to update the popper instance, e.g. create a new one, we don't want to trigger a component update, the only thing the child cares about is if the position in styles change, and that will be triggered when the new instance hits the updateStateModifier

I think I left the other version, with the force update to not break the original component API, in case they were public methods. I can make them more explicit and remove if not needed 👍

Reply:
Thanks I'll play with it on Monday once I hit the office.